splash/resources.py

```diff
@@ -105,8 +105,10 @@ def _writeOutput(self, data, request, content_type=None):
             data = json.dumps(data, cls=SplashJSONEncoder)
             return self._writeOutput(data, request, "application/json")
 
-        if isinstance(data, tuple) and len(data) == 2:
-            data, content_type = data
+        if isinstance(data, tuple) and len(data) == 3:
+            data, content_type, headers = data
+            for header in headers:
+                request.setHeader(str(header[0]), str(header[1]))
```

Comment:
What happens if non-ascii header values are passed? `str(header[0])` will encode them to bytes using the encoding which depends on user locale, etc. I think it is better to be more explicit and use either ascii or latin1 (I prefer ascii slightly more), and raise an error if a value can't be encoded.